mach

```diff
@@ -15,7 +15,16 @@
     run_in_nix_if_needed() {
         if { [ -f /etc/NIXOS ] || [ -n "${MACH_USE_NIX}" ]; } && [ -z "${IN_NIX_SHELL}" ]; then
             EXTRA_NIX_ARGS=${SERVO_ANDROID_BUILD:+'--arg buildAndroid true'}
+
+            # `nix-shell` needs the whole command passed as a single argument, so the arguments need
+            # to be shell-quoted. Rotate through the arguments, replacing them with quoted versions.
+            for arg in "$@"; do
+                set -- "$@" "$(printf \%q "$1")"
+                shift
+            done
+
             echo "NOTE: Entering nix-shell ${MACH_DIR}/shell.nix"
+            set -x  # Print the exact `nix-shell` command, quoted again so you can copy and paste.
```

Comment:
this is optional; we can remove it if it looks too noisy

Reply:
This does make things a bit noisy, but I'm fine to leave it in if this makes it easy for you to copy the exact nix-shell command in your workflow.

Reply:
thinking about it further, i’m not sure i actually ever need this. removed in 2493cfb7889405c221b8bd29581d4583cd1d91c9